src/components/competitions.jsx

```diff
@@ -6,11 +6,13 @@ import PastCompetitions from './pastCompetitions';
 const Competitions = () => {
 
     // sort competitions according to the end date and pass it to the respective components
+    const upComingCompetitons = competitions.filter(competition => new Date(competition.end_date) > new Date());
```

Comment:
For `upComingCompetitions` and `pastCompetitions`, please use `useMemo` as the `filter` can be expensive if the list is too big.